pkg/network/bootstrap.go

```diff
@@ -0,0 +1,39 @@
+package network
+
+import (
+	"github.com/pkg/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gophercloud/utils/openstack/clientconfig"
+
+	operv1 "github.com/openshift/api/operator/v1"
+)
+
+type KuryrBootstrapResult struct {
+	ServiceSubnet     string
+	PodSubnetpool     string
+	WorkerNodesRouter string
+	WorkerNodesSubnet string
+	PodSecurityGroups []string
+	ClusterID         string
+	OpenStackCloud    clientconfig.Cloud
+}
+
+type BootstrapResult struct {
+	Kuryr KuryrBootstrapResult
+}
+
+// Bootstrap creates resources required by SDN on the cloud.
+func Bootstrap(conf *operv1.NetworkSpec, client client.Client) (*BootstrapResult, error) {
+	switch conf.DefaultNetwork.Type {
+	case operv1.NetworkTypeKuryr:
+		return bootstrapKuryr(conf, client)
+	case operv1.NetworkTypeOpenShiftSDN:
+		return nil, nil
+	case operv1.NetworkTypeOVNKubernetes:
+		return nil, nil
+	}
+
+	return nil, errors.Errorf("unknown or unsupported NetworkType: %s", conf.DefaultNetwork.Type)
```

Comment:
We need to support unknown network types, so this should just be `return nil, nil`.